runtime/doc/ui.txt

```diff
@@ -576,6 +581,36 @@ because 'winhighlight' was used. UI items will be transmitted, even if the
 highlight group is cleared, so `ui_name` can always be used to reliably identify
 screen elements, even if no attributes have been applied.
 
+==============================================================================
+Details about UI element types				       *ui-elementtype*
+
+Activated by the `ext_elementtype` |ui-option|.
+Activates |ui-linegrid| implicitly.
+
+When this is set the `element_tags` parameter of |ui-event-hl_attr_define|
+contains semantic information about the type of UI element displayed.
```

Comment:
Why make it optional? An extra field in `hl_attr_define` is not a breaking change per `:help api-contract`, so we can just always emit it.

But I do like the leverage gained by [bfredl's suggestion](https://github.com/neovim/neovim/pull/28016#issuecomment-2018857464) to use highlight groups for this purpose instead.